pkg/api/validate/dynamic/dynamic.go

```diff
@@ -85,6 +90,14 @@ func NewValidator(log *logrus.Entry, env env.Interface, azEnv *azureclient.AROEn
 	}, nil
 }
 
+func NewServicePrincipalValidator(log *logrus.Entry, azEnv *azureclient.AROEnvironment, subscriptionID string, authorizerType AuthorizerType) (ServicePrincipalValidator, error) {
```

Comment:
```suggestion
func NewServicePrincipalValidator(log *logrus.Entry, azEnv *azureclient.AROEnvironment, authorizerType AuthorizerType) (ServicePrincipalValidator, error) {
```
We are not using the subscriptionID anywhere in this function.  I don't think this is necessary for the constructor.  

Is this missing the clients necessary to do the ValidateServicePrincipal check that it had when it was a full dynamic validator?

Reply:
This makes sense, disregard.

Reply:
No, @troy0820, I think you were right. `subscriptionID` is not used anywhere. It is passed into the function and then it dies.

Reply:
@nilsanderselde  The subscriptionID is passed into this constructor to fulfill the functions that create the rest of the dynamic type, it's just not explicitly set because this happens with functions below on the dynamic struct fields.
```
providers:          features.NewProvidersClient(azEnv, subscriptionID, authorizer),
spComputeUsage:     compute.NewUsageClient(azEnv, subscriptionID, authorizer),
spNetworkUsage:     network.NewUsageClient(azEnv, subscriptionID, authorizer),
permissions:        authorization.NewPermissionsClient(azEnv, subscriptionID, authorizer),
virtualNetworks:    newVirtualNetworksCache(network.NewVirtualNetworksClient(azEnv, subscriptionID, authorizer)),
diskEncryptionSets: compute.NewDiskEncryptionSetsClient(azEnv, subscriptionID, authorizer),
```
It doesn't need to be explicitly set because the subscriptionID is necessary for the above functions in the creation of this type.

Reply:
@m1kola ? Needs clarification